Sources/SmithyRetriesAPI/RetryError.swift

```diff
@@ -0,0 +1,11 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public enum RetryError: Error {
+    case maxAttemptsReached
+    case insufficientQuota
+}
```

Comment:
This type was originally declared along with the `DefaultRetryStrategy` and was internal.

Now, it's public and in the SmithyRetriesAPI module.

Reply:
Is this actually used by/part of SmithyRetriesAPI? I don't see mention of the 'quota' concept anywhere in API, but it is here (unless I've missed something). I also don't see it used anywhere, presumably its meant to be the type of error that could be thrown by `refreshTokenForRetry`?

Reply:
Also, might be a good time to add some more swift docs on what each of these types are, what they do, etc. (in general, not just for `RetryError`). Like RetryErrorInfo, RetryErrorType, RetryStrategy.